datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -470,6 +470,22 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 op: Or,
                 right,
             }) if is_false(&right) => *left,
+            // A OR !A ---> true (if A not nullable)
```

Comment:
Nice!  It's quit complicated deal with `null`.
in datafsuion:
```
❯ select true or NULL ;
Plan("'Boolean OR Null' can't be evaluated because there isn't a common type to coerce the types to")
❯ select true or FALSE ;
+---------------------------------+
| Boolean(true) OR Boolean(false) |
+---------------------------------+
| true                            |
+---------------------------------+
1 row in set. Query took 0.003 seconds.
❯
```
RUN in spark-sql
```
spark-sql> select true or false;
true
Time taken: 3.139 seconds, Fetched 1 row(s)
spark-sql> select true or NULL;
true
Time taken: 0.076 seconds, Fetched 1 row(s)
spark-sql> select FALSE or NULL;
NULL
Time taken: 0.06 seconds, Fetched 1 row(s)
spark-sql>
```
I think its ok get error when run  `NULL or xxx`

Reply:
👍  I suspect we would need to add some logic to https://github.com/apache/arrow-datafusion/blob/master/datafusion/optimizer/src/type_coercion.rs